tests/jobs/notifications/NewAnnouncementNotifyUsersTest.php

```diff
@@ -0,0 +1,128 @@
+    END;
+
+    /**
+     * Test job is a proper instance
+     */
+    public function testUnserializationGetProperJobInstance(): void
+    {
+        $this->assertInstanceOf(
+            NewAnnouncementNotifyUsers::class,
+            unserialize($this->serializedJobData)
+        );
+    }
+
+    /**
+     * Ensure that a serialized job can be unserialized and executed
+     */
+    public function testRunSerializedJob(): void
+    {
+        $this->mockMail();
+        
+        $this->mockRequest();
+
+        /** @var NewAnnouncementNotifyUsers $newAnnouncementNotifyUsersJob */
+        $newAnnouncementNotifyUsersJob = unserialize($this->serializedJobData);
+
+        $announcementMock = Mockery::mock(\PKP\announcement\Announcement::class)
+            ->makePartial()
+            ->shouldReceive([
+                'getAssocId' => 0,
+                'getLocalizedTitle' => '',
+            ])
+            ->withAnyArgs()
+            ->getMock();
+
+        $announcementRepoMock = Mockery::mock(app(AnnouncementRepository::class))
+            ->makePartial()
+            ->shouldReceive('get')
+            ->withAnyArgs()
+            ->andReturn($announcementMock)
+            ->getMock();
+        
+        app()->instance(AnnouncementRepository::class, $announcementRepoMock);
+
+        $contextDaoClass = get_class(Application::getContextDAO());
+
+        $contextMock = Mockery::mock(get_class(Application::getContextDAO()->newDataObject()))
+            ->makePartial()
+            ->shouldReceive([
+                'getId' => 0,
+            ])
+            ->withAnyArgs()
+            ->getMock();
+
+        $contextDaoMock = Mockery::mock($contextDaoClass)
+            ->makePartial()
+            ->shouldReceive('getById')
+            ->withAnyArgs()
+            ->andReturn($contextMock)
+            ->getMock();
+
+        DAORegistry::registerDAO(substr(strrchr($contextDaoClass, '\\'), 1), $contextDaoMock);
```

Comment:
I think I'd prefer a match statement on e.g. `Application::getName` rather than string parsing and assembly. At some point we'll get rid of `JournalDAO`, for example, and being able to grep for it will be helpful.